PR for SRVCOM-3678: Update the Serverless 1.36 release notes (#93729)
openshift/openshift-docs: modules/serverless-rn-1-36-0.adoc

    variables,
    state,
    lastUpdate,
    nodes {
      name
    }
}
----

* {ServerlessLogicProductName} Data Index now supports filtering queries using workflow definition metadata. Additionally, Data Index has been updated to consume deploy and undeploy events, improving the accuracy and traceability of workflow state. 

* {ServerlessLogicOperatorName} now emits events to the Data Index to indicate when a workflow definition becomes available or unavailable. These deploy and undeploy events are sent automatically when the Data Index is present in the same namespace. 

[id="fixed-issues-1-36-0_{context}"]
== Fixed issues

* Previously, the Knative Kafka dispatcher could stop consuming events if a Kafka consumer group rebalance occurred while a sink was processing events out of order. This behavior triggered errors such as:

** `SEVERE: Unhandled exception`
** `java.lang.IndexOutOfBoundsException: bitIndex < 0`
** Repeated logs like `Request joining group due to: group is already rebalancing`

This issue is now fixed. The dispatcher correctly handles out-of-order event consumption during rebalancing and continues processing events without interruption.

> * Previously, a KafkaSource remained in a `Ready` state even when `KafkaSource.spec.net.tls.key` failed to load due to the use of unsupported TLS certificates in PKCS #1 format. This issue is now fixed. An appropriate error is now reported when attempting to create a KafkaBroker, KafkaChannel, KafkaSource, or KafkaSink using TLS certificates in an unsupported format.

Reviewer: Should the following instances be in back ticks? +1 to ticks if these are CR names
- KafkaBroker, KafkaChannel, KafkaSource, KafkaSink

Author: Yes, you are right! They should be in back ticks.

[id="known-issues-1-36-0_{context}"]
== Known issues

* The `openshift-serverless-1/logic-swf-builder-rhel8:1.35.0` image currently downloads persistence extensions from Maven during the build process. 

* If the `swf-dev-mode` image is started with a broken or invalid workflow definition, the container may enter a stuck state.

* The delivery of the Maven repository in the `devmode` image is not yet automated for offline environments.

> * When deploying a workflow in the `preview` profile on {ocp-product-title}, if the initial build fails and is later corrected, the Operator does not create the corresponding workflow deployment. As a result, the deployment remains missing and the SonataFlow status is not updated, even after the build is fixed.

Reviewer: ```suggestion
* When deploying a workflow in the `preview` profile on {ocp-product-title}, if the initial build fails and is later corrected, the Operator does not create the corresponding workflow deployment. As a result, the deployment remains missing and the `SonataFlow` status is not updated, even after the build is fixed.
```

Should use the ticks right?

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-36-0_{context}"]
= Red Hat {ServerlessProductName} 1.36

{ServerlessProductName} 1.36 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:

[id="new-features-1-36-0_{context}"]

> == New features

Author: @ricardozanini I have made the changes accordingly. Thank you!

* {ServerlessProductName} now uses Knative Serving 1.16.
* {ServerlessProductName} now uses Knative Eventing 1.16.
* {ServerlessProductName} now uses Kourier 1.16.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.16.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.16.
* The `kn func` CLI plugin now uses `func` 1.16.

* `IntegrationSource` and `IntegrationSink` are now available as a Technology Preview. These are Knative Eventing custom resources that support selected Kamelets from the Apache Camel project. Kamelets enables you to connect to third-party systems for improved connectivity, acting as either sources (event producers) or sinks (event consumers).

* Knative Eventing can now automatically discover and register EventTypes based on the structure of incoming events. This feature simplifies the configuration and management of EventTypes, reducing the need for manual definitions. This feature is available as a Technology Preview.

> * {ServerlessProductName} Eventing introduces EventTransform, a new API resource that allows you to declaratively transform JSON events without writing custom code. With EventTransform, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. EventTransform integrates seamlessly with Knative sources, triggers, and brokers, enhancing interoperability in event-driven architectures.

Reviewer 2: EventTransform should also be Tech Preview (please update it here and create an entry in the TP table) 

(see the PM call notes)  CC: @matzew

[id="fixed-issues-1-36-0_{context}"]
== Fixed issues

* Previously, the Knative Kafka dispatcher could stop consuming events if a Kafka consumer group rebalance occurred while a sink was processing events out of order. This behavior triggered the following errors:

** `SEVERE: Unhandled exception`
** `java.lang.IndexOutOfBoundsException: bitIndex < 0`
** Repeated logs like `Request joining group due to: group is already rebalancing`

+

This issue is now fixed. The dispatcher correctly handles out-of-order event consumption during rebalancing and continues processing events without interruption.

* Previously, a KafkaSource remained in a `Ready` state even when `KafkaSource.spec.net.tls.key` failed to load due to the use of unsupported TLS certificates in PKCS #1 format. This issue is now fixed. An appropriate error is now reported when attempting to create a `KafkaBroker`, `KafkaChannel`, `KafkaSource`, or `KafkaSink` using TLS certificates in an unsupported format.

[id="known-issues-1-36-0_{context}"]
== Known issues

> * The `openshift-serverless-1/logic-swf-builder-rhel8:1.35.0` image currently downloads persistence extensions from Maven during the build process. 

Reviewer 3: ```suggestion
* The `openshift-serverless-1/logic-swf-builder-rhel8:1.35.0` and `openshift-serverless-1/logic-swf-builder-rhel8:1.36.0` images currently downloads persistence extensions from Maven during the build process. 
```

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-36-0_{context}"]
= Red Hat {ServerlessProductName} 1.36

{ServerlessProductName} 1.36 is now available. New features, updates, fixed issues, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:

[id="new-features-1-36-0_{context}"]
== New features

[id="new-features-eventing-1-36-0_{context}"]
=== {ServerlessProductName} Eventing

* {ServerlessProductName} now uses Knative Eventing 1.16.

* {ServerlessProductName} now uses Knative for Apache Kafka 1.16.

* `IntegrationSource` and `IntegrationSink` are now available as a Technology Preview. These are Knative Eventing custom resources that support selected Kamelets from the Apache Camel project. Kamelets enables you to connect to third-party systems for improved connectivity, acting as either sources (event producers) or sinks (event consumers).

> * Knative Eventing can now automatically discover and register EventTypes based on the structure of incoming events. This feature simplifies the configuration and management of EventTypes, reducing the need for manual definitions. This feature is available as a Technology Preview.

Reviewer 4: According to the Knative docs, `EventType` is a Kubernetes object. 
```suggestion
* Knative Eventing can now automatically discover and register `EventTypes` objects based on the structure of incoming events. This feature simplifies the configuration and management of event types, reducing the need for manual definitions. This feature is available as a Technology Preview.
```

> * {ServerlessProductName} Eventing introduces `EventTransform`, a new API resource that you can use to declaratively transform JSON events without writing custom code. With `EventTransform`, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. `EventTransform` integrates seamlessly with Knative sources, triggers, and brokers, enhancing interoperability in event-driven architectures. This feature is now available as a Technology Preview.

Reviewer 4: I see in the Knative docs that they use EventTransform without mark-up or _object_. I don't think we should, and should follow our standard 
```suggestion
* {ServerlessProductName} Eventing introduces `EventTransform` objects, a new API resource that you can use to declaratively transform JSON events without writing custom code. With `EventTransform` objects, you can modify attributes, extract or reshape data, and streamline event flows across systems. Common use cases include event enrichment, format conversion, and request-response transformation. Event transformations integrate seamlessly with Knative sources, triggers, and brokers, enhancing interoperability in event-driven architectures. This feature is now available as a Technology Preview.
```

+

> See the following key features of `EventTransform`:

Reviewer 4: ```suggestion
See the following key features of `EventTransform` objects:
```
or 
```suggestion
See the following key features of event transformations:
```

[id="fixed-issues-1-36-0_{context}"]
== Fixed issues

[id="fixed-issues-eventing-1-36-0_{context}"]
=== {ServerlessProductName} Eventing

* Previously, the Knative Kafka dispatcher could stop consuming events if a Kafka consumer group rebalance occurred while a sink was processing events out of order. This behavior triggered the following errors:

** `SEVERE: Unhandled exception`
** `java.lang.IndexOutOfBoundsException: bitIndex < 0`
** Repeated logs like `Request joining group due to: group is already rebalancing`

+

This issue is now fixed. The dispatcher correctly handles out-of-order event consumption during rebalancing and continues processing events without interruption.

* Previously, a KafkaSource remained in a `Ready` state even when `KafkaSource.spec.net.tls.key` failed to load due to the use of unsupported TLS certificates in PKCS #1 format. This issue is now fixed. An appropriate error is now reported when attempting to create a `KafkaBroker`, `KafkaChannel`, `KafkaSource`, or `KafkaSink` using TLS certificates in an unsupported format.

[id="known-issues-1-36-0_{context}"]
== Known issues

[id="known-issues-osl-1-36-0_{context}"]
=== {ServerlessLogicProductName}

> * If the `swf-dev-mode` image is started with a broken or invalid workflow definition, the container might enter a stuck state.

Reviewer 4: Is _stuck_ state the proper term?
